File: frontend/src/app/services/dashboard/dashboard.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { DashboardCuentaCorrienteResponse, ReportFilter } from '../../core/models/models';

/**
 * Servicio HTTP para los KPIs del dashboard.
 * Llama a transacciones-service via api-gateway BFF.
 */
@Injectable({ providedIn: 'root' })
export class DashboardService {

  private apiUrl = `${environment.apiGateway}/bff/dashboard`;

  constructor(private http: HttpClient) {}

  /**
   * Obtiene los KPIs de cuenta corriente del dashboard.
   * Soporta filtros opcionales por obra, cliente, proveedor y rango de fechas.
   *
   * @param filtro ReportFilter con parámetros opcionales
   * @returns Observable de DashboardCuentaCorrienteResponse con: cobrado, porCobrar, pagado, porPagar, resultado
   */
  getCuentaCorriente(filtro: ReportFilter): Observable<DashboardCuentaCorrienteResponse> {
    return this.http.post<DashboardCuentaCorrienteResponse>(
      `${this.apiUrl}/cuenta-corriente`,
      filtro
    );
  }
}

File: backend1.0/transacciones-service/src/main/java/com/transacciones/service/DashboardService.java
package com.transacciones.service;

import com.transacciones.dto.DashboardCuentaCorrienteResponse;
import com.transacciones.dto.DashboardFilterRequest;
import com.transacciones.dto.TopObraFinancieroDto;
import java.util.List;

/**
 * Servicio para KPIs del dashboard.
 */
public interface DashboardService {

  /**
   * Obtiene los KPIs de cuenta corriente del dashboard con filtros opcionales.
   * Ejecuta sp_dashboard_cuenta_corriente en la base de datos.
   *
   * @param filtro DTO con parámetros opcionales: obraId, clienteId, proveedorId, fechaInicio, fechaFin
   * @return DashboardCuentaCorrienteResponse con: cobrado, porCobrar, pagado, porPagar, resultado
   */
  DashboardCuentaCorrienteResponse obtenerCuentaCorriente(DashboardFilterRequest filtro);

  List<TopObraFinancieroDto> obtenerTopObras(int topN);
}
